Cython/Utility/NumpyImportArray.c

```diff
@@ -18,3 +18,27 @@ if (unlikely(_import_array() == -1)) {
 }
 #endif
 #endif
+
+///////////////////////// NumpyImportUFunc.init ////////////////////
+
+// Unlike import_array, this is generated by the @cython.ufunc decorator
+// so we're confident the right headers are present and don't need to override them
+
+{
+    // NO_IMPORT_UFUNC is Numpy's mechanism for indicating that import_umath is handled elsewhere
+#if !NO_IMPORT_UFUNC /* https://numpy.org/devdocs/reference/c-api/ufunc.html#c.NO_IMPORT_UFUNC */
+    if (unlikely(_import_umath() == -1)) {
+        PyErr_SetString(PyExc_ImportError, "numpy.core.umath failed to import "
+        "(auto-generated by @cython.ufunc).");
+    }
+#endif
```

Comment:
```suggestion
#else
    if ((0)) {}
#endif
```

It's possible to generate invalid code with `NO_IMPORT_UFUNC` so just cleaning this up